### DISCUSSION TITLE
Copilot Enterprise Mixed Licensing (Beta)

### DISCUSSION THREAD

**Discussion by goldieowner17** [community/community#132831] [Copilot News and Announcements]
We're pleased to announce the public beta of [Copilot mixed licensing](https://github.blog/changelog/2024-07-26-copilot-enterprise-mixed-licensing-beta) for enterprise customers.

Enterprise customers can now select a Copilot plan per organization, allowing them to adopt Copilot Business and Copilot Enterprise according to each organization's needs and timelines.

We welcome your feedback!

**Comment by mcurole** [comment]
When will we be able to mix licensing within an organization? This feature doesn't help us at all as we have a single large organization.

**Comment by casey-ford** [comment]
@goldieowner17 We currently have Copilot for Business assigned to users in a single organization. To test our Copilot Enterprise with a small group of Engineers, do you suggest creating a separate org for Copilot Enterprise and giving access to the subset of Engineers?

**Reply by casey-ford** [reply]
It was recommended to me by our GitHub Account team to set this up in the following way.

1. Create an organization strictly for licensing Copilot for Business
2. Add the users you want to have Copilot for Business to this organization, preferably using an IdP. Assign a Copilot for Business license to these users. 
3. In your main GitHub organization, enable Copilot Enterprise. Assign a Copilot Enterprise license to individuals in this org. Features like Copilot Knowledgebases require the source code to be in the same organization where Copilot Enterprise is enabled. Also, GitHub dedups the license allocation so you won't be charged for 2 licenses on a user who has been assigned both licenses.

**Reply by srodgers-awaze** [reply]
That's an interesting solution - I'll have a look at if that's possible without too much disruption in our account, cheers!

**Reply by jsskarthik7** [reply]
Did it work as expected?

**Comment by srodgers-awaze** [comment]
Just adding to the other voices here, ideally this would also be available at an individual or group level _within_ an organisation too - we don't want to upgrade all of our teams to copilot enterprise at once, and splitting out teams into seperate orgs doesn't make sense for us either.

**Comment by Trip-T** [comment]
Team, quick question, what happens when a user is added to 2 different orgs and one has Business and other enterprise please? will this count towards 2 licenses applied for the same user? or Does the enterprise license only apply and its counted only once?

**Reply by srodgers-awaze** [reply]
According to the information @casey-ford was given, GitHub internally de-dupes the licenses if a user has both licenses assigned.

**Reply by Trip-T** [reply]
Thanks @srodgers-awaze , so Enterprise pricing will apply and the license capabilities as well?

**Reply by srodgers-awaze** [reply]
It looks like that's that's the case - Enterprise will supercede Copilot for Business and you'll only be billed for the Enterprise license.

@goldieowner17 Can you confirm so that it's "official"?

**Comment by Nk974** [comment]
> We're pleased to announce the public beta of [Copilot mixed licensing](https://github.blog/changelog/2024-07-26-copilot-enterprise-mixed-licensing-beta) for enterprise customers.
> 
> 
> Enterprise customers can now select a Copilot plan per organization, allowing them to adopt Copilot Business and Copilot Enterprise according to each organization's needs and timelines.
> 
> We welcome your feedback!

**Comment by gidich** [comment]
Question -

Is GitHub looking to add coding agent to the GitHub CoPilot Business subscription, we currently use GitHub Teams and would like to use the Copilot coding agent however it looks to be locked out from us.
